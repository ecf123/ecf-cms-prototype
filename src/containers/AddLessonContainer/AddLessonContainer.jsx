import "./AddLessonContainer.scss";
import { useState } from "react";
import Short from "../../components/Short/Short";
import SelectComponent from "../../components/SelectComponent/SelectComponent";
import Button from "../../components/Button/Button";
import MediaUploadBox from "../../components/MediaUploadBox/MediaUploadBox";
import AddLessonAdditionalBoxes from "../../components/AddLessonAdditionalBoxes/AddLessonAdditionalBoxes";
import QuestionAnswer from "../../components/QuestionAnswer/QuestionAnswer";
import QuestionContainer from "../../containers/QuestionContainer/QuestionContainer";
import LessonContentPreview from "../../components/LessonContentPreview/LessonContentPreview";
import { useNavigate } from "react-router-dom";
import PageTitle from "../../components/PageTitle/PageTitle";

const AddLessonContainer = () => {
  const [inputs, setInputs] = useState(true);
  const [data, setData] = useState([]);
  let navigate = useNavigate();
  const [additionalInfo, setAdditionalInfo] = useState({
    title: "",
    content: "",
  });
  const [question] = useState({ question: "", answers: [] });

  const onChange = (event) => event.target.value === "Lesson" ? setInputs(true) : setInputs(false);

  const [pageInfo, setPageInfo] = useState({
    file: null,
    fileName: "",
    shortInputOne: "",
    shortInputTwo: "",
    questions: [],
    additionalInfos: []
  });

  const handleFileChange = (e) => {
    if (!pageInfo.file) setPageInfo({ fileName: e.target.files[0].name });
    pageInfo.file
      ? setPageInfo({ ...pageInfo, file: null })
      : setPageInfo({
          ...pageInfo,
          file: URL.createObjectURL(e.target.files[0]),
        });
  };

  const addData = () => {
    setData([...data, additionalInfo]);
  };

  const handleDelete = (e) => {
    inputs
      ? setData(
          data.filter(
            (object) =>
              object.content !=
              e.target.parentElement.parentElement.children[1].firstChild.data
          )
        )
      : setData(
          data.filter(
            (object) =>
              object.question !=
              e.target.parentElement.parentElement.children[0].firstChild.childNodes[2].data
          )
        );
  };
  let array = [];
  const addAnswers = (e) => {
    array.push(e.target.parentElement.children[3][1].value);
    array.push(e.target.parentElement.children[3][2].value);
    array.push(e.target.parentElement.children[3][3].value);
    array.push(e.target.parentElement.children[3][4].value);
    setData([
      ...data,
      {
        ...question,
        question: e.target.parentElement.children[3][0].value,
        answers: array,
      },
    ]);
  };

  const handleSubmit = () => {
    console.log(pageInfo)
    navigate("/asd") // back to lessons
  }

  const handleCancel = () => {
    navigate("/xczc") // back to lessons
  }

  return (<>
    <div className="lesson__header">
    <PageTitle
      className="lesson__title page-title"
      title="Add Lesson"
    />
  </div>
    <div className="lesson">
      <div className="lesson__container">
        <Short
          shortLabelText="Lesson Name"
          shortType="text"
          handleShortValue={(e) =>
            setPageInfo({ ...pageInfo, shortInputOne: e.target.value })
          }
          inputClassName="short__input"
          name="lesson name"
        />
        <SelectComponent className="lesson__select" onChange={onChange} />
        <Short
          shortLabelText="Estimated Completion Time"
          shortType="text"
          handleShortValue={(e) =>
            setPageInfo({ ...pageInfo, shortInputTwo: e.target.value })
          }
          inputClassName="short__input"
          name="estimated completion time"
        />
        {inputs ? (
          <AddLessonAdditionalBoxes
            handleFreeTypeValue={(e) => {
              setAdditionalInfo({ ...additionalInfo, content: e.target.value });
            }}
            handleShortValue={(e) =>
              setAdditionalInfo({ ...additionalInfo, title: e.target.value })
            }
          />
        ) : (
          <QuestionAnswer onSubmit={addAnswers}></QuestionAnswer>
        )}
        ;
        <h5
          className="lesson__add"
          onClick={inputs ? addData : addAnswers}
        >
          Add +
        </h5>
        <div className="lesson__buttons">
          <Button
            style={"button light-grey round-border large"}
            textStyle={"text large-text light-grey"}
            buttonText={"Cancel"}
            buttonFunction={handleCancel}
          />
          <Button
            style={"button black round-border large"}
            textStyle={"text large-text black"}
            buttonText={"Submit"}
            buttonFunction={handleSubmit}
          />
        </div>
      </div>
      <div className="add-lesson__upload">
        <MediaUploadBox
          file={pageInfo.file}
          handleFileChange={handleFileChange}
          uploadLabelName="Lesson Upload"
          uploadButtonText="Media Upload"
          fileName={pageInfo.fileName}
        />
      </div>
      <div className="lesson__content">
        {inputs ? (
          <LessonContentPreview
            lessonsArray={data}
            handleDelete={handleDelete}
          />
        ) : (
          <QuestionContainer
            questionsArray={data}
            handleDelete={handleDelete}
          />
        )}
        
        <button className="lesson__add-button" onClick={() => inputs ? setPageInfo({ ...pageInfo, additionalInfos: data }) : setPageInfo({ ...pageInfo, questions: data })}>Confirm data</button>
      </div>      

    </div>
    </>
  );
};

export default AddLessonContainer;
