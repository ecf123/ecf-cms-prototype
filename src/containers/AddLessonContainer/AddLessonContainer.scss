@use "../../styles/_variables.scss" as *;

.lesson {
  display: flex;
  gap: 2.18rem;
  margin-top: 3rem;
  margin-left: 3rem;

  &__buttons {
    display: flex;
    flex-direction: row;
    gap: 0.56rem;
    margin-top: 1rem;
  }
  &__add {
    text-decoration: underline dotted;
    text-underline-offset: 0.25rem;
    color: $color-blue-navy-dark;
    font-family: $font-secondary;
    font-style: normal;
    font-weight: 500;
    font-size: 0.8rem;
    cursor: pointer;
  }
  &__container {
    display: flex;
    flex-direction: column;
    gap: 1.68rem;
  }
  &__content {
    height: 684px;
    width: 430px;
  }

  &__add-button {
    margin-top: 2.5rem;
    padding: 1rem 2.5rem;
    background: #2d303a;
    border: none;
    border-radius: 2.5rem;
    cursor: pointer;
    &:active {
      opacity: 0.8;
    }
  }

  &__header {
    padding: 0 3rem 3rem 3rem;
    border-bottom: 2px $color-grey-light solid;
    display: block;
  }
}
