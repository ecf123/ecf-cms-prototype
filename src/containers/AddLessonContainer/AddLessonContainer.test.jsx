import { customRender } from "../../utils/testUtils";
import AddLessonContainer from "./AddLessonContainer";

it("it should match the snapshot", () => {
    const { container } = customRender(
      <AddLessonContainer  />
    );
    expect(container).toMatchSnapshot();
  });