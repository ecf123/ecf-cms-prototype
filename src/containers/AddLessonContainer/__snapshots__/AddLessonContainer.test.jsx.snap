// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot 1`] = `
<div>
  <div
    class="lesson__header"
  >
    <h1
      class="lesson__title page-title"
    >
      Add Lesson
    </h1>
  </div>
  <div
    class="lesson"
  >
    <div
      class="lesson__container"
    >
      <div
        class="short"
      >
        <label
          class="short__label"
        >
          Lesson Name
        </label>
        <input
          class="short__input"
          name="lesson name"
          spellcheck="true"
          type="text"
        />
      </div>
      <div>
        <label
          class="select__label"
        >
          Lesson Type
        </label>
        <div
          class="select"
        >
          <select
            class="select__dropdown"
            src="upside-down-arrow-grey.svg"
          >
            <option
              class="select__dropdownoption"
            >
              Lesson
            </option>
            <option
              class="select__dropdownoption"
            >
              Assessment Quiz
            </option>
          </select>
        </div>
      </div>
      <div
        class="short"
      >
        <label
          class="short__label"
        >
          Estimated Completion Time
        </label>
        <input
          class="short__input"
          name="estimated completion time"
          spellcheck="true"
          type="text"
        />
      </div>
      <div
        class="add-lesson"
      >
        <div
          class="short"
        >
          <label
            class="short__label"
          >
            Additional Subtitle(s) (optional)
          </label>
          <input
            class="short__input"
            name="input-one"
            placeholder="e.g. Fund Manager"
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          class="free-type"
        >
          <label
            class="free-type__label"
          >
            Additional Content (optional)
          </label>
          <textarea
            class="free-type__input"
            name="input-two"
            spellcheck="true"
            type="text"
          />
        </div>
      </div>
      ;
      <h5
        class="lesson__add"
      >
        Add +
      </h5>
      <div
        class="lesson__buttons"
      >
        <button
          class="button light-grey round-border large"
        >
          <div />
          <p
            class="text large-text light-grey"
          >
            Cancel
          </p>
        </button>
        <button
          class="button black round-border large"
        >
          <div />
          <p
            class="text large-text black"
          >
            Submit
          </p>
        </button>
      </div>
    </div>
    <div
      class="add-lesson__upload"
    >
      <div
        class="thumbnail"
      >
        <label
          class="thumbnail__label"
        >
          Lesson Upload
        </label>
        <div
          class="thumbnail__container"
        >
          <div
            class="thumbnail__btn--wrapper"
          >
            <button
              class="button light-grey round-border small"
            >
              <div />
              <p
                class="text small-text light-grey"
              >
                Media Upload
              </p>
            </button>
            <input
              accept="image/*,video/*,audio/*,.txt"
              class="thumbnail__btn--input"
              type="file"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="lesson__content"
    >
      <div
        class="lesson-preview"
      />
      <button
        class="lesson__add-button"
      >
        Confirm data
      </button>
    </div>
  </div>
</div>
`;
